eth/api_backend.go

```diff
@@ -343,3 +348,61 @@ func (b *EthAPIBackend) StatesInRange(ctx context.Context, fromBlock *types.Bloc
 func (b *EthAPIBackend) StateAtTransaction(ctx context.Context, block *types.Block, txIndex int, reexec uint64) (core.Message, vm.BlockContext, *state.StateDB, func(), error) {
 	return b.eth.stateAtTransaction(block, txIndex, reexec)
 }
+
+// AccessList creates an access list for the given transaction.
+// If the accesslist creation fails an error is returned.
+// If the transaction itself fails, an vmErr is returned.
+func (b *EthAPIBackend) AccessList(ctx context.Context, block *types.Block, reexec uint64, args *ethapi.SendTxArgs) (acl types.AccessList, gasUsed uint64, vmErr error, err error) {
+	if block == nil {
+		block = b.CurrentBlock()
+	}
+	db, release, err := b.eth.stateAtBlock(block, reexec)
+	defer release()
+	if err != nil {
+		return nil, 0, nil, err
+	}
+	if err := args.SetDefaults(ctx, b); err != nil {
+		// most likely an error in gas estimation, set gas to max gas
+		log.Warn("Setting defaults failed", "error", err)
+		maxGas := hexutil.Uint64(b.RPCGasCap())
+		args.Gas = &maxGas
+	}
+	var (
+		gas        uint64
+		accessList types.AccessList
+		msg        types.Message
+	)
+	if args.AccessList != nil {
+		accessList = *args.AccessList
+	}
+	for i := 0; i < 10; i++ {
```

Comment:
Wut? What is this loop here?

Reply:
We need to rerun the transaction with the new accesslist in case that unlocks more slots/addresses